AlfiyaSiddique/TastyTrails
# Share recipie option

## What does this PR do?


This PR solves the issue #152  . Here i have added backend code so that when user clicks share button automatically number of shares for that recipe gets updated by one. After that i update number of share on client side by using UseState. I don't want to make unnecessary call to backend so i increase it by one. But when page is loaded again fresh data has updated number of shares. As share button is pressed I access current web url and add id of the post at last which becomes url of paticular recipe. It is copied in clipboard and user can share it. When web is hosted it will give correct url becuase it will have proper domain. I have also updated that page that displays individual Recipe. In that page state was passed from all Recepie page but if we try to access that using individual url then it will cause error as state was not passed at that time. So i updated code that when there is no state then it should request backend for content of that page using id of individual recipe. This will help in case if state is not passed or any user tries to access individual recipe using url that has been shared to him. I have also added feature that when user click share button he gets option to go to different social media like whatsapp ,facebook where he can share these thing.

If you don't want the last feature then i can simply copy to clipboard and show that to user that it has been copied. Please mention whether you want the last feature or not. 

i have installed react icons for generating icons of different social media.

Fixes # (issue)
Fixes #152 

![Screenshot 2024-10-21 193812](https://github.com/user-attachments/assets/8bac1bd4-0048-41f1-8d6b-730d12b2a447)
![Screenshot 2024-10-21 193832](https://github.com/user-attachments/assets/d51378e7-5739-4c39-b28a-b3a233304af5)

## How should this be tested?
Click on the share button and see the changes. Run the copied url in another tab. 


I tested responsiveness of my component and shared the recipie. I checked database it was properly updated. I tested url in another chrome on the same device.
Please note currently it is copying localhost because i was running it on  my localhost if user will run it on hosted website he will be able to capture url of that website and it will work perfectly fine. I didn't hard code it i have used window.location so that i can capture actual website url and used routing to properly direct to any page so that if any user visits the url he can see that.


- Test 1
- Test 2

## Checklist



### Required

- [x ] Filled out the "How to test" section in this PR
- [x ] Read Contributing guide
- [x ] Self-reviewed my own code
- [x ] Checked for warnings, there are none
- [ x] Removed all `console.logs`
- [x ] The changes don't cause any responsiveness issues

Requst: Can you please add level 3 if you feel i have worked as per that
